web/frontend/src/pages/election/components/ElectionForm.tsx

```diff
@@ -1,39 +1,166 @@
-import React, { FC } from 'react';
+import React, { ChangeEvent, FC, useCallback, useState } from 'react';
 import PropTypes from 'prop-types';
 import { useTranslation } from 'react-i18next';
+import { CloudUploadIcon } from '@heroicons/react/outline';
 
 import { ENDPOINT_EVOTING_CREATE } from 'components/utils/Endpoints';
 
+import { Configuration, Rank, Select, Subject, Text } from '../../../components/utils/types';
+
+import AddButton from './AddButton';
+import SubjectComponent from './SubjectComponent';
+
+import { getObjSubject } from './utils/getObjectType';
+import UploadFile from './UploadFile';
+
 type ElectionFormProps = {
   setShowModal(modal: any): void;
   setTextModal(text: string): void;
 };
 
 const ElectionForm: FC<ElectionFormProps> = ({ setShowModal, setTextModal }) => {
   const { t } = useTranslation();
+  const [schema, setSchema] = useState<Configuration>({ MainTitle: '', Scaffold: [] });
+  const { MainTitle, Scaffold } = schema;
 
   async function createHandler() {
     const data = {
       Title: '',
       AdminID: '',
       Token: '',
-      Format: '',
+      Format: schema,
     };
 
     const req = {
       method: 'POST',
       type: 'application/json',
       body: JSON.stringify(data),
     };
-    const res = await fetch(ENDPOINT_EVOTING_CREATE, req);
-    if (res.status !== 200) {
-      console.warn('failed to create election');
+    try {
+      const res = await fetch(ENDPOINT_EVOTING_CREATE, req);
+      if (res.status !== 200) {
+        setTextModal('Failed to create election');
```

Comment:
The response here according to the API documentation is `ElectionID`, so the response body only contains the property `response.ElectionID`. So at best I think that the modal text includes this ID to display for the user (with some text before), right ?

Reply:
If a 200 happens then yes, you'll get the `ElectionID`. Otherwise the server might send a body containing some error text. This is what `http.Error` does: https://github.com/dedis/d-voting/blob/f57713cbf1cd33c21b0df652bc6ec514da493476/contracts/evoting/controller/handlers.go#L44